pom.xml

```diff
@@ -2737,7 +2739,7 @@
             <reportsDirectory>${project.build.directory}/surefire-reports</reportsDirectory>
             <junitxml>.</junitxml>
             <filereports>SparkTestSuite.txt</filereports>
-            <argLine>-ea -Xmx4g -Xss4m -XX:MaxMetaspaceSize=2g -XX:ReservedCodeCacheSize=${CodeCacheSize} -Dio.netty.tryReflectionSetAccessible=true</argLine>
+            <argLine>-ea -Xmx5g -Xss4m -XX:MaxMetaspaceSize=2g -XX:ReservedCodeCacheSize=${CodeCacheSize} ${extraJavaTestArgs} -Dio.netty.tryReflectionSetAccessible=true</argLine>
```

Comment:
Does this mean JDK17 require more memory? Otherwise, please revert this.

Reply:
Yes, use -Xmx4g, `SPARK-36464: size returns correct positive number even with over 2GB data` will oom

Reply:
That sounds like a blocker for JDK17 adoption. Could you file a JIRA instead? Otherwise, this will bite us in the future.

Reply:
Our Jenkins job has failed for a long time:
https://amplab.cs.berkeley.edu/jenkins/job/spark-master-test-maven-hadoop-2.7-jdk-11/

```
*** RUN ABORTED ***
  java.lang.OutOfMemoryError: Java heap space
  at java.base/java.nio.HeapByteBuffer.<init>(HeapByteBuffer.java:61)
  at java.base/java.nio.ByteBuffer.allocate(ByteBuffer.java:348)
  at org.apache.spark.util.io.ChunkedByteBufferOutputStreamSuite.$anonfun$new$23(ChunkedByteBufferOutputStreamSuite.scala:125)
  at org.apache.spark.util.io.ChunkedByteBufferOutputStreamSuite.$anonfun$new$23$adapted(ChunkedByteBufferOutputStreamSuite.scala:125)
  at org.apache.spark.util.io.ChunkedByteBufferOutputStreamSuite$$Lambda$9510/0x00000008427c4840.apply(Unknown Source)
  at org.apache.spark.util.io.ChunkedByteBufferOutputStream.allocateNewChunkIfNeeded(ChunkedByteBufferOutputStream.scala:87)
  at org.apache.spark.util.io.ChunkedByteBufferOutputStream.write(ChunkedByteBufferOutputStream.scala:75)
  at java.base/java.io.OutputStream.write(OutputStream.java:122)
  at org.apache.spark.util.io.ChunkedByteBufferOutputStreamSuite.$anonfun$new$22(ChunkedByteBufferOutputStreamSuite.scala:127)
  at org.apache.spark.util.io.ChunkedByteBufferOutputStreamSuite$$Lambda$1002/0x0000000840913840.apply(Unknown Source)
```

Reply:
Create SPARK-36900

Reply:
Yes, go ahead and file a JIRA to raise this to something like 8g just to be safe. I'd also double the stack size to 8m. I have run into this too. It's separate, but worth changing.